src/main/java/uk/gov/companieshouse/charges/data/config/ExceptionHandlerConfig.java

```diff
@@ -30,21 +38,53 @@ public ExceptionHandlerConfig(final Logger logger) {
     @ExceptionHandler(value = {Exception.class})
     public ResponseEntity<Object> handleException(Exception ex, WebRequest request) {
         var correlationId = generateShortCorrelationId();
-        logger.error(String.format("Started: handleException: %s Generating error response for "
-                        + "Exception: % "
-                        + "Cause: %"
-                        + "StackTrace: % ",
-                correlationId, ex.getClass().toString(), ex.getCause().toString(),
-                ex.getStackTrace().toString()));
+        logger.error(String.format("Started: handleException: %s Generating error response ",
+                correlationId), ex);
         Map<String, Object> responseBody = new LinkedHashMap<>();
-        responseBody.put("timestamp", LocalDateTime.now());
-        responseBody.put("message", "There is issue completing the request.");
-        responseBody.put("correlationId", correlationId);
+        populateResponseBody(responseBody, correlationId);
         request.setAttribute("javax.servlet.error.exception", ex, 0);
         logger.error(String.format("Finished: handleException: %s handleException", correlationId));
+        if (ex instanceof ResponseStatusException){
+            ResponseStatusException rse = (ResponseStatusException)ex;
+            Throwable cause = rse.getCause();
+            if (cause instanceof IOException){
+                return new ResponseEntity(responseBody, HttpStatus.SERVICE_UNAVAILABLE);
+            }
+            if ("invokeChsKafkaApi".equals(rse.getReason())){
+                return new ResponseEntity(responseBody, HttpStatus.NOT_EXTENDED);
+            }
+        }
         return new ResponseEntity(responseBody, HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
+    @ExceptionHandler(value = {ResponseStatusException.class})
+    public ResponseEntity<Object> handleException(ResponseStatusException ex, WebRequest request) {
+        var correlationId = generateShortCorrelationId();
+        logger.error(String.format("Started: handleException: %s Generating error response ",
+                correlationId), ex);
+        Map<String, Object> responseBody = new LinkedHashMap<>();
+        populateResponseBody(responseBody, correlationId);
+        Throwable cause = ex.getCause();
+        if (cause instanceof IOException){
```

Comment:
Can we have a separate IOException please

Reply:
No, the IO exception is wrapped by the ResponseStatusException as the exception thrown and thus caught by the exception handler.